src/handle.c

```diff
@@ -45,11 +45,17 @@ static uv_handle_t* luv_check_handle(lua_State* L, int index) {
   // we get its metatable and check that we get `true` back
   // when looking the metatable up in the "uv_handle" table.
   lua_getfield(L, LUA_REGISTRYINDEX, "uv_handle");
-  lua_getmetatable(L, index < 0 ? index - 1 : index);
+  isHandle = lua_getmetatable(L, index < 0 ? index - 1 : index);
+  if (!isHandle) goto fail;
   lua_rawget(L, -2);
   isHandle = lua_toboolean(L, -1);
   lua_pop(L, 2);
-  if (isHandle) { return handle; }
+  if (!isHandle) goto fail;
+  // cast the uesrdata to uv_handle_t
+  handle = *(uv_handle_t**)udata;
+  if (!handle || !handle->data) goto fail;
```

Comment:
Nitpick: I believe this `!handle` check is redundant, as the `*(uv_handle_t**)udata` cast cannot make `handle` null and we already know `udata` is not null.

```suggestion
  if (!handle->data) goto fail;
```